### PR TITLE
cloudflared: add livecheck

### PR DESCRIPTION
**Pull request by Omoeba**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
What was the problem with the livecheck before you added this?

**Comment by samford** [comment]
The `livecheck` block in this PR doesn't do anything differently than the default check (i.e., without a `livecheck` block), so it's not needed at this point. The only functional difference is that the presence of a `livecheck` block would cause `brew livecheck` to skip the built-in unstable version filtering, which isn't something we want/need to do in this particular context. This also doesn't need a regex to restrict matching yet (the default `Git` strategy regex is working fine so far), so it's preferable to simply rely on the default check instead of adding the unnecessary `livecheck` block in this PR.

In the future, please use `brew livecheck --debug` to better understand livecheck's behavior for a given formula before creating a PR to add/update a `livecheck` block. In this case, the debug output shows that livecheck already checks the stable URL, preprocesses it into the Git repository URL, and then successfully identifies versions from tags (implicitly using the default strategy regex). In this particular context, you can also use `git ls-remote --tags https://github.com/cloudflare/cloudflared.git` to list the tags in the repository (e.g., to understand if we need to add a regex to restrict matching to certain tag formats), as this is what the `Git` strategy uses internally.

If you create additional livecheck PRs in the future, please explain the reasoning behind adding/updating a `livecheck` block in the PR description (e.g., what isn't working in the existing check, what the PR fixes, etc.), as this helps us to understand the changes.

That said, I appreciate the effort but we should pass on this.

**Comment by Omoeba** [comment]
> The `livecheck` block in this PR doesn't do anything differently than the default check (i.e., without a `livecheck` block), so it's not needed at this point. The only functional difference is that the presence of a `livecheck` block would cause `brew livecheck` to skip the built-in unstable version filtering, which isn't something we want/need to do in this particular context. This also doesn't need a regex to restrict matching yet (the default `Git` strategy regex is working fine so far), so it's preferable to simply rely on the default check instead of adding the unnecessary `livecheck` block in this PR.
> 
> In the future, please use `brew livecheck --debug` to better understand livecheck's behavior for a given formula before creating a PR to add/update a `livecheck` block. In this case, the debug output shows that livecheck already checks the stable URL, preprocesses it into the Git repository URL, and then successfully identifies versions from tags (implicitly using the default strategy regex). In this particular context, you can also use `git ls-remote --tags https://github.com/cloudflare/cloudflared.git` to list the tags in the repository (e.g., to understand if we need to add a regex to restrict matching to certain tag formats), as this is what the `Git` strategy uses internally.
> 
> If you create additional livecheck PRs in the future, please explain the reasoning behind adding/updating a `livecheck` block in the PR description (e.g., what isn't working in the existing check, what the PR fixes, etc.), as this helps us to understand the changes.
> 
> That said, I appreciate the effort but we should pass on this.

I did run `brew livecheck --debug` and `Livecheckable?:   No` was included in the output. Is there no practical difference between `Livecheckable?:   No` and `Livecheckable?:   Yes`?

**Comment by samford** [comment]
The `Livecheckable?` boolean in the debug output (and the related [`#livecheckable?` method](https://rubydoc.brew.sh/Formula.html#livecheckable%3F-class_method)) indicate whether a `livecheck` block is present in a given formula/cask. This should be read as "Is there a `livecheck` block?" not "Is `brew livecheck` able to check this?".

Whether `brew livecheck` works for a formula is primarily indicated by whether it returns the correct latest stable version. We generally only add a `livecheck` block if `brew livecheck` doesn't correctly identify the latest stable version or we otherwise need to manually control or override some aspect of livecheck's behavior (e.g., to only match appropriate versions, etc.).

That said, I agree that the "livecheckable" language can be confusing. This terminology is a holdover from years ago when livecheck's data was kept in the separate Homebrew/homebrew-livecheck repository. Back in the day, formula-specific livecheck data was kept in a `Livecheckables` folder, where the filenames corresponded to the related formulae filenames (e.g., `a52dec.rb`). These files simply consisted of a class (same name as the related formula class) with a `#livecheck` method call inside. A file like this was called a "livecheckable", so that's why `livecheckable?` indicates whether a `livecheck` block is present for a given formula.

Originally, the `#livecheck` method took a hash (e.g., here's an old [`a52dec.rb` livecheckable](https://github.com/Homebrew/homebrew-livecheck/blob/9f4cfc9f684163ba8a88cb3aa50820ace4597b9d/Livecheckables/a52dec.rb)) but we re-implemented this as a `livecheck` block in the `Formula` DSL before migrating this data into formulae. It wasn't until after this that we stopped using the "livecheckable" terminology and instead referred to it as simply a `livecheck` block but this is one area where the older language hasn't been updated yet.

Long story short, it would make more sense to update this to something like `livecheck block?:` (along with a `livecheck_block?` or `has_livecheck_block?` method) but it will take a little bit of tweaking within livecheck to accomplish this (i.e., it's not _quite_ as simple as it appears on the surface). It's on my to-do list but I'm working through some other tasks at the moment. In the interim time, I hope this helps to clarify this particular point.

**Comment by Omoeba** [comment]
Thanks for clarifying!
